src/Voron/Impl/SliceSmallSet.cs

```diff
@@ -60,38 +66,36 @@ private IEnumerable<TValue> ReturnValues()
 
         public unsafe void Add(Slice key, TValue value)
         {
-            if (_overflowStorage != null)
-            {
-                goto Overflow;
-            }
-
             int idx = FindKey(key);
-            if (idx == Invalid)
+            if (idx != Invalid)
+                goto Done;
+
+            // side effect, overflow if needed
+            idx = RequestWritableBucket();
+            if (idx == Invalid || _overflowStorage != null)
             {
-                // side affect, overflow if needed
-                idx = RequestWritableBucket();
-                if (idx == Invalid)
-                    goto Overflow;
+                Debug.Assert(_overflowStorage != null);
+                _overflowStorage[key] = value;
             }
```

Comment:
Should be `return` here,no?

Reply:
No longer. One change that I did is to treat the list as a LIFO cache instead of discarding it completely as it was originally. That simplifies a lot of the code.